packages/broker/test/unit/plugins/operator/MaintainTopologyHelper.test.ts

```diff
@@ -0,0 +1,202 @@
+import { MaintainTopologyHelper } from '../../../../src/plugins/operator/MaintainTopologyHelper'
+import { ContractFacade } from '../../../../src/plugins/operator/ContractFacade'
+import { mock, MockProxy } from 'jest-mock-extended'
+import { Provider } from '@ethersproject/providers'
+import EventEmitter3 from 'eventemitter3'
+import { eventsWithArgsToArray, randomEthereumAddress } from '@streamr/test-utils'
+import { toStreamID } from '@streamr/protocol'
+import { wait } from '@streamr/utils'
+
+const fromArray = async function* <T>(items: T[]): AsyncGenerator<T> {
+    for (const item of items) {
+        yield item
+    }
+}
+
+const STREAM_ONE_ID = toStreamID('streamOne')
+const STREAM_TWO_ID = toStreamID('streamTwo')
+const SPONSORSHIP_ONE = randomEthereumAddress()
+const SPONSORSHIP_TWO = randomEthereumAddress()
+
+describe(MaintainTopologyHelper, () => {
+    let smartContractEventEmitter: EventEmitter3
+    let contractFacade: MockProxy<ContractFacade>
+    let helper: MaintainTopologyHelper
+
+    async function emitSmartContractEvent(event: 'Staked' | 'Unstaked', sponsorship: string): Promise<void> {
+        smartContractEventEmitter.emit(event, sponsorship)
+        await wait(0)
```

Comment:
Why `wait` is needed?

Reply:
The event handler for event `Staked` is async in `MaintainTopologyHelper`. A pause occurs when the handler gets the stream id for the sponsorship from the contractFacade. Therefore we need to pause here to allow for the event handler to go thru.

We don't technically need `wait(0)` when the event is `Unstaked` but I don't think it is worth adding an if-clause for this.

Reply:
Could add a comment about the wait (or rename the function e.g. to `triggerEventHandler`)